### PR TITLE
feat(cockpit): manage cockpit installation registration

### PR DESCRIPTION
**Pull request by phiz71**
* creation of an *Installation* in database
* creation of a command producer to register to cockpit
* creation of an API resource to fetch the installation information

Closes gravitee-io/issues#4766

### CODE REVIEW

File: gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get installation information",
+            notes = "User must have the INSTALLATION[READ] permission on the platform")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Installation definition", response = InstallationEntity.class),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+    @Permissions({
+            @Permission(value = RolePermission.ORGANIZATION_SETTINGS, acls = { READ })
```
Comment (aelamrani) [inline gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java:59]: Should be INSTALLATION / READ no?

File: gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/InstallationServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        assertEquals(INSTALLATION_ID, installationEntity.getId());
+        assertEquals(now, installationEntity.getCreatedAt());
+        assertEquals(now, installationEntity.getUpdatedAt());
+        assertEquals(2, installationEntity.getAdditionalInformation().size());
+        assertEquals(COCKPIT_INSTALLATION_ID, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_ID));
+        assertEquals(COCKPIT_INSTALLATION_STATUS, installationEntity.getAdditionalInformation().get(InstallationService.COCKPIT_INSTALLATION_STATUS));
+    }
+
+    @Test
+    public void shouldUpdateAdditionalInformation() throws TechnicalException {
```
Comment (aelamrani) [inline gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/InstallationServiceTest.java:131]:
```
Tests run: 7, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.021 sec <<< FAILURE! - in io.gravitee.rest.api.service.InstallationServiceTest
shouldUpdateAdditionalInformation(io.gravitee.rest.api.service.InstallationServiceTest)  Time elapsed: 0.005 sec  <<< FAILURE!
java.lang.AssertionError: null
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at io.gravitee.rest.api.service.InstallationServiceTest.shouldUpdateAdditionalInformation(InstallationServiceTest.java:151)
```

**Review by aelamrani — CHANGES_REQUESTED** [review]

File: gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import static io.gravitee.rest.api.model.permissions.RolePermissionAction.READ;
+
+/**
+ * Defines the REST resources to manage Portal.
```
Comment (brasseld) [inline gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java:38]:
```suggestion
 * Defines the REST resources to manage an installation.
```

File: gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+@Path("/installation")
+public class InstallationResource {
+
+    @Inject
+    private InstallationService installationService;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get installation information",
+            notes = "User must have the INSTALLATION[READ] permission on the platform")
```
Comment (brasseld) [inline gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java:53]: Are you sure about `INSTALLATION` ?
It looks to be `ORGANIZATION_SETTINGS` according to the annotation

**Review by brasseld — CHANGES_REQUESTED** [review]

File: gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java (hunk 1 above)
Comment (phiz71) [reply gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java]: Fixed

File: gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/InstallationServiceTest.java (hunk 2 above)
Comment (phiz71) [reply gravitee-rest-api-service/src/test/java/io/gravitee/rest/api/service/InstallationServiceTest.java:131]: It was hard to reproduce. I've changed the way I compare two dates.

File: gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java (hunk 4 above)
Comment (phiz71) [reply gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java:53]: The permission was not the right one. It has been fixed. (See https://github.com/gravitee-io/gravitee-management-rest-api/pull/1344#discussion_r562710308)

File: gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java (hunk 3 above)
Comment (phiz71) [reply gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/installation/InstallationResource.java]: Fixed

**Review by aelamrani — APPROVED** [review]

**Pull request merged**